Weatherman.py

```diff
@@ -0,0 +1,277 @@
+    maximum1 = 0
+    maximum2 = 0
+    for iterator in range(0, len(temperature_values)):
+        if temperature_values[iterator] != '':
+            if float(temperature_values[iterator]) >= maximum1:
+                maximum1 = float(temperature_values[iterator])
+                maximum2 = iterator
+            else:
+                temperature_values[iterator] = temperature_values[iterator]
+    return maximum2
+
+
+def calculate_minimum_temperature(temperature_values):
+    minimum1 = float(temperature_values[0])
+    minimum2 = 0
+    for iterator in range(1, len(temperature_values)):
+        if temperature_values[iterator] != '':
+            if float(temperature_values[iterator]) < minimum1:
+                minimum1 = float(temperature_values[iterator])
+                minimum2 = iterator
+            else:
+                minimum1 = minimum1
+    return minimum2
+
+
+def display_results(maximum_temperature, maximum_date, minimum_temperature, minimum_date,
+                    mean_humidity, mean_humidity_date):
+    print(f'Max temperature is {maximum_temperature}C on 'f'{change_date_format(maximum_date)}')
+    print(f'Min temperature is {minimum_temperature}C on 'f'{change_date_format(minimum_date)}')
+    print(f'Max Humidity is {mean_humidity}% on 'f'{change_date_format(mean_humidity_date)}')
+
+
+def change_date_format(date):
+    date_format = date.split('-')
+    month_number = date_format[1]
+    datetime_object = datetime.strptime(month_number, "%m")
+    month_name = datetime_object.strftime("%b")
+    new_date = f'{date_format[2]}' + f' {month_name} '
+    return new_date
+
+
+def find_month_name(number):
+    datetime_object = datetime.strptime(number, "%m")
+    month_name = datetime_object.strftime("%b")
+    return month_name
+
+
+def run_weather_man(argument):
+    if argument.year != '':
+        year = argument.year.split('/')
+        if len(year) == 1:
+            run_weather_man_year(argument.year)
+            print('')
+        else:
+            print('Wrong Input')
+    if argument.year_month != '':
+        year_charts = argument.year_month.split('/')
+        if len(year_charts) == 2:
+            run_weather_man_month(year_charts[0], find_month_name(year_charts[1]), '-c')
+            print('')
+        else:
+            print('Wrong Input')
+    if arguments.year_and_month != '':
+        year_month = argument.year_and_month.split('/')
+        if len(year_month) == 2:
+            run_weather_man_month(year_month[0], find_month_name(year_month[1]), '-a')
+            print('')
+        else:
+            print('Wrong Input')
+
+# Main
+
+
+if __name__ == '__main__':
+    White = "\033[37m"
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('-e', '--year', action='store', default='')
+    parser.add_argument('-c', '--year_month', action='store', default='')
+    parser.add_argument('-a', '--year_and_month', action='store', default='')
```

Comment:
We need to take decisions on the base of our arguments i.e. `e`, `a` or `c` so no need to define them with year month, and so on. Simply consume them and work accordinbgly